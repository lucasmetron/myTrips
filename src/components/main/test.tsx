import React from 'react'
import { render, screen } from '@testing-library/react'
import Main from './index' //por ele esta na pasta, pode se colocar somente um .

//serve para descreve o que vamos começar a testar
describe('<Main />', () => {
  //it serve para escrever o teste de fato
  it('should rende the heading', () => {
    //Este teste esta renderizar o componente e espera encontrar um heading escrito 'react avançado'
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avançado/i }) //o i buscar maiusculas e minuscula
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
