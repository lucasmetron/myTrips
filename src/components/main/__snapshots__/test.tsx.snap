// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should rende the heading 1`] = `
<main>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
